### DISCUSSION TITLE
Handle externally managed files, while also encrypting them.

### DISCUSSION THREAD

**Discussion by saharshbhansali** [twpayne/chezmoi#3828] [Q&A] [answered]
I've already read [this](https://www.chezmoi.io/user-guide/manage-different-types-of-file/#handle-configuration-files-which-are-externally-modified) section of the user guide, which details how to handle externally managed files. But I'm not sure if I can achieve this while also encrypting the data present within it.

What I want to manage is a (chromium-based) browser profile. This contains both configuration settings, as well as extensions and other private data. Thus, I was considering encrypting the entire browser profile. 

Is there any way I can achieve doing this with chezmoi? If not, are there any other alternatives to it? 

At the very least, I would like to manage the browser settings. But the best case is still to manage everything related to the browser profile.

**Comment by twpayne** [comment]
No, this is not really possible as the external application is not aware of the encryption and so will corrupt the encrypted files.

If you want encryption and externally managed files then you'll have to update them in chezmoi's source state (e.g. using `chezmoi merge-all` or `chezmoi re-add`) every time the external application modifies them.

**Reply by saharshbhansali** [reply]
Then, is there a way I can encrypt and add the files to the repository using the encryption I've already set up in chezmoi but also ignore it while running commands like `chezmoi apply`? This would make it so that I can periodically re-add the files to keep the settings and configurations up to date, without having to figure out an entirely different system to actually encrypt them after ignoring them in chezmoi outright (by using `.chezmoiignore`).

**Reply by twpayne** [reply, marked as answer]
> Then, is there a way I can encrypt and add the files to the repository using the encryption I've already set up in chezmoi but also ignore it while running commands like `chezmoi apply`?

Yes, you can run

    chezmoi apply --exclude=encrypted
